File: SortingAlgorithms/insertionSort.java
package SortingAlgorithms;

public class insertionSort {
    public static void main(String[] args) {

        int[] myArray = {64, 34, 25, 12, 22, 11, 2, 5};

        for (int i=1; i<myArray.length; i++) {
            int insertIndex = i;
            int currentValue = myArray[i];
            int j = i - 1;

            while (j>=0 && myArray[j] > currentValue) {
                myArray[j + 1] = myArray[j];
                insertIndex = j;
                j--;
            }
            myArray[insertIndex] = currentValue;
        }

        System.out.print("Sorted array: ");
        for (int value : myArray) {
            System.out.print(value + " ");
        }
    }
}

File: SortingAlgorithms/insertion_sort.py
# implementation of insertion sort algorithm.

sample_array = list(map(int, input().split()))
length = len(sample_array)

for i in range(1, length):
    insert_index = i
    current_value = sample_array.pop(i)
    for j in range(i-1, -1, -1):
        if sample_array[j] > current_value:
            insert_index = j
    sample_array.insert(insert_index, current_value)

print(sample_array)